frontend/src/static/js/v_zoom.js

```diff
@@ -72,10 +111,52 @@ window.vZoom = {
     },
 
     restoreZoomTab() {
-      // restore selected user's commits from v_summary
+      // restore selected user's commits and file type colors from v_summary
       this.$root.$emit('restoreCommits', this.info);
     },
 
+    updateFileTypes(commits) {
+      this.fileTypes = [];
+      commits.forEach((day) => {
+        day.commitResults.forEach((slice) => {
+          Object.keys(slice.fileTypesAndContributionMap).forEach((fileType) => {
+            if (this.fileTypes.indexOf(fileType) === -1) {
+              this.fileTypes.push(fileType);
+            }
+          });
+        });
+      });
+      this.fileTypes.sort();
+      this.updateSelectedFileTypes();
+    },
+
+    updateSelectedFileTypes() {
+      this.retrieveSelectedFileTypesHash();
+      if (this.selectedFileTypes === undefined) {
+        this.selectedFileTypes = this.fileTypes.slice();
+      }
```

Comment:
the undefined case will only occur at initialization? then why not put it in filteredUser and then combine the rest of updateSelectedFileTypes with retrieveSelectedFileTypesHash